Python/codegen.c

```diff
@@ -792,6 +792,17 @@ codegen_process_deferred_annotations(compiler *c, location loc)
         return SUCCESS;
     }
 
+    instr_sequence *old_instr_seq = INSTR_SEQUENCE(c);
+    instr_sequence *nested_instr_seq = NULL;
```

Comment:
Is there a reason why we can't reorder the code in this function so that we first emit the instructions for annotations and then for the code (so we don't need to do this injection)?

Reply:
Ah I see, it's stuff from outside this function.

Reply:
Yeah, I considered an alternative where we generate the code first, then later edit the LOAD_CONST instruction to the correct code object we generate later, but that seemed like it'd be worse.

Reply:
I wonder if we could merge the annotation code into the main code in `_PyCfg_OptimizedCfgToInstructionSequence`.

Reply:
Or rather, `_PyCfg_FromInstructionSequence`.

Reply:
Like we'd put some special marker in the instruction sequence and then have `_PyCfg_FromInstructionSequence` stitch it together? I'm not sure that would be simpler.

Reply:
We can build two instruction sequences here, and when it comes time to build the CFG we use both as inputs. 

We could have a pseudo-instruction inserted after the RESUME of a module, which means "insert the annotations code here". Then if the instruction sequence for annotations is non-NULL, _PyCfg_FromInstructionSequence injects it.

Reply:
I'll try it out but still not convinced that would be better than what I have here.

Reply:
Done! Wasn't too bad after all.

Reply:
Cool. So now we shouldn't need `_PyCompile_SetInstrSequence` and `nested_instr_seq`, right?

Reply:
I see. I think I would have moved this into the compiler. Instead of adding `_PyCompile_SetInstrSequence`, add functions to "begin annotations block" and "end annotations block" and let the compiler manage its data structures by itself.

Reply:
Done